### ISSUE TITLE
x86_64-pc-windows-msvc compile error

### ISSUE THREAD

**Issue by ijl** [Alexhuszagh/rust-lexical#26]
I've encountered the following compile error on a `vs2017-win2016` VM (in Azure Piplines) using `x86_64-pc-windows-msvc`, `rustc 1.39.0-nightly (97e58c0d3 2019-09-20)`. I see this project has a CI run on `x86_64-pc-windows-gnu` (not `msvc`) that passes.

```
error[E0061]: this function takes 1 parameter but 0 parameters were supplied
   --> C:\Users\VssAdministrator\.cargo\registry\src\github.com-1ecc6299db9ec823\lexical-core-0.6.1\src\util\num.rs:961:44
    |
961 |         float_method_msvc!(self, f32, f64, powf, powf32, n as f32)
    |                                            ^^^^ expected 1 parameter
```

**Comment by ijl** [comment]
lexical-core 0.4.5 compiles ok on the same target.

**Comment by Alexhuszagh** [comment]
@ijl, interesting, since this code hasn't been modified in a long time (since almost when I started the project, you can see the [same code](https://github.com/Alexhuszagh/rust-lexical/blob/9591df153d2882341547195f23c373999aaeb975/lexical-core/src/util/num.rs#L914) back in December 2018, before I even publicly announced it). I may have to add MSVC CI for this.

**Comment by Alexhuszagh** [comment]
@ijl Can you check the latest commit fixes the issue?

**Comment by ijl** [comment]
@Alexhuszagh, thanks for looking at it. I tried `lexical-core = { git = "https://github.com/Alexhuszagh/rust-lexical.git", rev = "8f6cb83e5795c38019d5922b2124f35ce55943ed",  features = ["correct", "ryu", "unchecked_index"]}` and had:

```
error[E0308]: mismatched types
   --> C:\Users\VssAdministrator\.cargo\git\checkouts\rust-lexical-fa5b35bbaa4489f8\8f6cb83\lexical-core\src\util\num.rs:961:58
    |
961 |         float_method_msvc!(self, f32, f64, powf, powf32, n as f32)
    |                                                          ^^^^^^^^
    |                                                          |
    |                                                          expected f64, found f32
    |                                                          help: you can convert an `f32` to `f64`: `(n as f32).into()`

error: could not compile `lexical-core`.
```

**Comment by Alexhuszagh** [comment]
@ijl I guess my `target_environment = "msvc"` doesn't seem to actually be checking this code branch, so Travis CI might not be sufficient. Can you check the latest commit, I believe logically I've patched this.
0e0e50fe9ef434a24f97485420d0756aed7c16fd

**Comment by ijl** [comment]
@Alexhuszagh 0e0e50f compiles and passes tests. Thank you.

**Comment by Alexhuszagh** [comment]
Patched as of v0.6.2 and v0.4.6, both of which are published on crates.io. Thanks for reporting this.
